### PR TITLE
ft: ARSN-84 Switch to Jest

### PR DESCRIPTION
**Pull request by ghivert**
Switch the test runner to Jest, and upgrade ESLint.
All files are committed in order to comply with the new ESLint.
The test runner is now Jest and the code coverage is Jest + Istanbul.
All new test files should be written as `[filename].spec.js` or `[filename].spec.ts`.

### CODE REVIEW

**Comment by jbertran** [comment]
Reviewers: please pay special attention to `2f326fdee` and `93f4ae2`.

**Comment by jbertran** [comment]
Regarding code coverage:

Looking at the diff, what is reported as coverage changes is just down to differences in how mocha and jest detect coverage of a given line.

File: package.json [hunk 1]
```diff
@@ -54,10 +55,11 @@
   },
   "devDependencies": {
     "@sinonjs/fake-timers": "^6.0.1",
-    "eslint": "2.13.1",
+    "eslint": "^8.9.0",
     "eslint-config-airbnb": "6.2.0",
     "eslint-config-scality": "scality/Guidelines#ec33dfb",
     "eslint-plugin-react": "^4.3.0",
+    "jest": "^27.5.1",
     "mocha": "8.0.1",
```
Comment (miniscruff) [inline package.json]: can we remove mocha now?

Comment (jbertran) [reply package.json]: Oops. We can indeed, fixed.

**Review by tmacro — APPROVED** [review]

**Review by miniscruff — APPROVED** [review]

**Comment by ghivert** [comment]
Fixed

**Pull request merged**